src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser, User } from './App.types';
+import { TodoList } from './components/TodoList';
+
+const todosWithUser: TodoWithUser[] = todosFromServer.map(
+  (todo): TodoWithUser => {
+    const user = usersFromServer.find(({ id }) => id === todo.userId) as User;
+
+    return {
+      ...todo,
+      user,
+    };
+  },
+);
+const defaultUserId = '0';
+const titleRegexp = /^[a-zA-Zа-яА-ЯёЁїЇіІєЄґҐ0-9\s]*$/;
+
+const validateTitle = (title: string): boolean => titleRegexp.test(title);
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>([...todosWithUser]);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(defaultUserId);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+  const isNotSelectedUser = userId === defaultUserId;
+  const hasForbiddenSymbolsInTitle = !validateTitle(title);
+  const titleErrorMessage = hasForbiddenSymbolsInTitle
+    ? 'Title contains special symbols'
+    : 'Please enter a title';
+
+  const clearForm = () => {
+    setTitle('');
+    setUserId(defaultUserId);
+  };
+
+  const handleSubmit = (
+    event: React.FormEvent<HTMLFormElement>,
+  ): null | void => {
+    event.preventDefault();
+    if (!title || hasForbiddenSymbolsInTitle) {
+      setHasTitleError(true);
+    }
+
+    if (isNotSelectedUser) {
+      setHasUserError(true);
+    }
+
+    if (title && !isNotSelectedUser) {
+      const user = usersFromServer.find(({ id }) => id === +userId) as User;
```

Comment:
converting to a number with `+` is not a mistake, but not considered a clean approach. To make it easier to understand we usually convert the type explicitly, like `Number(userId)`